### PR TITLE
feat(@desktop/wallet): Wallet accounts - Account List

### LINKED ISSUE #11189: Wallet accounts - Account List
https://www.figma.com/file/idUoxN7OIW2Jpp3PMJ1Rl8/%E2%9A%99%EF%B8%8F-Settings-%7C-Desktop?type=design&node-id=15248-306907&t=gjr7i8Cks7sph3BX-4

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
fixes #11189

### What does the PR do

New wallet account list in profile view 

https://www.figma.com/file/idUoxN7OIW2Jpp3PMJ1Rl8/%E2%9A%99%EF%B8%8F-Settings-%7C-Desktop?type=design&node-id=15248-306907&t=gjr7i8Cks7sph3BX-4

### Affected areas

Wallet Settings Page

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

https://github.com/status-im/status-desktop/assets/60327365/3ac387a3-dca2-47e0-9d49-a7913b29206c

### CODE REVIEW

File: ui/app/AppLayouts/Profile/controls/WalletAccountDelegate.qml [hunk 1]
```diff
@@ -1,21 +1,30 @@
...
 
 import utils 1.0
 
 StatusListItem {
     id: root
     
     property var account
-    property bool showShevronIcon: true
+    property string chainShortNames
+    property var totalCount: 0
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/controls/WalletAccountDelegate.qml:17]: `int`?

File: ui/app/AppLayouts/Profile/controls/WalletAccountDelegate.qml [hunk 2]
```diff
@@ -1,21 +1,30 @@
...
     signal goToAccountView()
-    
-    title: account.name
-    subTitle: account.address
+
     objectName: account.name
-    asset.color: Utils.getColorForId(account.colorId)
+    title: account.name
+    subTitle: {
+        let elidedAddress = StatusQUtils.Utils.elideText(account.address,6,4)
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/controls/WalletAccountDelegate.qml:24]: `const elidedAddress`

File: ui/app/AppLayouts/Profile/controls/WalletAccountDelegate.qml [hunk 3]
```diff
@@ -24,16 +33,22 @@ StatusListItem {
...
         icon: "next"
         color: Theme.palette.baseColor1
     }
+
+    onClicked: goToAccountView()
+
+    // This is used to give the first and last delgate rounded corners
+    Rectangle {
+        visible: totalCount > 1
+        property bool isLastOrFirstItem: index === 0 || index === (totalCount-1)
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/controls/WalletAccountDelegate.qml:46]: `readonly`

File: ui/app/AppLayouts/Profile/controls/WalletKeyPairDelegate.qml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+    signal goToAccountView(var account)
+
+    QtObject {
+        id: d
+        property var relatedAccounts: model.keyPair.accounts
+        property bool isWatchOnly: model.keyPair.pairType === Constants.keycard.keyPairType.watchOnly
+        property bool isPrivateKeyImport: model.keyPair.pairType === Constants.keycard.keyPairType.privateKeyImport
+        property bool isProfileKeypair: model.keyPair.pairType === Constants.keycard.keyPairType.profile
+        property string locationInfo: model.keyPair.migratedToKeycard ? qsTr("On Keycard"): qsTr("On device")
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/controls/WalletKeyPairDelegate.qml:25]: `readonly` all those?

File: ui/app/AppLayouts/Profile/controls/WalletKeyPairDelegate.qml [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+                charactersLen: 2
+                isLetterIdenticon: !model.keyPair.icon && !asset.name.toString()
+            }
+            components: [
+                StatusFlatRoundButton {
+                    icon.name: "more"
+                    icon.color: Theme.palette.directColor1
+                    visible: !d.isWatchOnly
+                },
+                StatusBaseText {
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/controls/WalletKeyPairDelegate.qml:64]: `StatusSwitch` can take a text, move it there directly as `text: foo`

**Review by caybro — COMMENTED** [review]
Looks very nice, some minor comments inline

File: ui/app/AppLayouts/Profile/controls/WalletKeyPairDelegate.qml (hunk 5 above)
Comment (Khushboo-dev-cpp) [reply ui/app/AppLayouts/Profile/controls/WalletKeyPairDelegate.qml:64]: the text assigns itself to the right not left of the checkbox 

image

**Review by alaibe — APPROVED** [review]

File: ui/app/AppLayouts/Profile/controls/WalletKeyPairDelegate.qml (hunk 5 above)
Comment (Khushboo-dev-cpp) [reply ui/app/AppLayouts/Profile/controls/WalletKeyPairDelegate.qml:64]: @caybro Ive confirmed with Ben and the text should be to the left of the checkbox. Please approve when you can :)

**Review by caybro — APPROVED** [review]
LGTM!

File: src/app/modules/shared_models/keypair_item.nim [hunk 6]
```diff
@@ -9,6 +9,7 @@ type
     Profile
     SeedImport
     PrivateKeyImport
+    WatchOnly
```
Comment (saledjenic) [inline src/app/modules/shared_models/keypair_item.nim:12]: @Khushboo-dev-cpp this refers to a keypair type, there is no watch only keypair, that's the point and the reason why I didn't want to add it. It logically doesn't belong to this enum.

Could we change this somehow?

**Pull request merged**